src/FakeItEasy/DummyFactory.cs

```diff
@@ -31,20 +31,19 @@ public bool CanCreate(Type type)
         /// Creates the dummy.
         /// </summary>
         /// <param name="type">The type of dummy to create.</param>
-        /// <returns>The dummy object.</returns>
+        /// <returns>The dummy object. Unlike a dummy provided by built-in FakeItEasy mechanisms, may be <c>null</c>.</returns>
         /// <exception cref="ArgumentException"><paramref name="type"/> is not <typeparamref name="T"/>.</exception>
-        public object Create(Type type)
+        public object? Create(Type type)
         {
             this.AssertThatFakeIsOfCorrectType(type);
 
-            return this.Create(); // this might be null. Same with any implementation of IDummyFactory.
-            // We could allow it and trap in DynamicDummyFactory, returning false there.
+            return this.Create();
         }
 
         /// <summary>
         /// Creates the dummy.
         /// </summary>
-        /// <returns>The dummy object.</returns>
+        /// <returns>The dummy object. Unlike a dummy provided by built-in FakeItEasy mechanisms, may be <c>null</c>.</returns>
         protected abstract T Create();

```

Comment:
Or, since I am not a fan of the `null`s, someone who wants this badly enough can implement the interface. Mwu-ha-ha!

Reply:
Fair point! On the other hand, the doc does say the method can return null, so it should probably allow it... Otherwise, we should remove the comment in the doc

Reply:
Yeah, I was mostly kidding. And of course we'd update the docs to align with code and policy.
You made good points about allowing custom Dummies that are nulls. I think we might as well continue down the path.

Reply:
Oh. But if the client has the warnings on, and wants to return a null, should their factory instead be a `DummyFactory<ReferenceType?>`? That quells the warning about returning null...

Reply:
I guess I'm assuming warnings _and_ language features on. Nullability is hard!

Reply:
I tried adding `[return: MaybeNull]`, but the warning stayed. I'm probably doing something foolish.